### DISCUSSION TITLE
NFTs Idea: Non-transferable option?

### DISCUSSION THREAD

**Discussion by tijno** [deso-protocol/dips#59] [Medium CIPs]
I think currently all NFTs can be resold by the owner.

Would it be worth adding:

* the option to make an NFT non tradable / non sellable ? (note: 100% royalties kinda makes it non sellable)
* allowing the NFT minter to gift 1 NFT of a 1 of Many to user accounts?

Doing this would open up new and innovative solutions to solve common problems around:

1 - communities
2 - badgers & achievements
3 - qualifications / skills

**After reviewing the code, some more thoughts about implementation approach:**

* Making a NFT "Unsellable" or "Non transferable" I think will require an additional param in the Transaction, similar to `IsForSale` - eg `CantBeSold` set to `false` by default, but `true` if NFT is minted with the `non transferable` option enabled.
* Looks like NFTs can be updated to handle i guess the reselling of them. Its important the `CantBeSold` option can not be changed as a buyer or winner of such a non-transferable NFT should not be able to change this.
* `BidAmount` can be 0 in `AcceptNFTBid` which is good so this could be used as a basis for gifting/awarding a NFT to a user
* Gifting would still need an additional function as I think `AcceptNFTBid` checks against bids submitted.

**Purpose**

With these 2 features, the following would be able to use NFTs as a mechanism:

* Rewards & achievements - eg DH's challebe for the pulse dashboard could have been turned into a 1 of 1 NFT badge gifted to the winner
* Challenges - The Pulse Trading Challenge, or the Dev Challenge I did.
* Open communities / interest groups - Lets say I want to manage a community of developers on Bitclout, I could mint a 1 of many non transferable NFT and gift those to developers that have contributed to Bitclout
* Closed communities - Closed 100 seats only community using NFTs to identify its members and give the members assurances that only 100 members will ever have access.

**To roll this out fully I think the following would be needed too:**

- Could use `100%` royalties as a way of making an NFT non-sellable. 
- `CantBeSold` setting for each NFT as mentioned above
- Gift or transfer NFT to a user - but probably only allow this if this was not won in a Auction. Otherwise it would offer a way for users to sell NFTs without royalties.
- Ability to return a gifted NFT as per [this post](https://bitclout.com/posts/d85d56024dfe24578795bca3ce07f62fe03132b88019543d71e7649b51e39f3b) by (SeanSlater)[https://bitclout.com/u/SeanSlater]
- Show "award badge" on user profile. Could use NFT tab for this, which shows the NFTs owned by the user (`GetNFTsForUser` function already exists to do this)
- Get a list of users that own a certain NFT - would require a `GetUsersForNFT`
- For Gated Content, have api that returns `true` if User owns a specific NFT as per [this post](https://bitclout.com/posts/d85d56024dfe24578795bca3ce07f62fe03132b88019543d71e7649b51e39f3b) by (SeanSlater)[https://bitclout.com/u/SeanSlater]

**Comment by omicreativedev** [comment]
YES VOTE

**Reply by tijno** [reply]
Thx. I think its possible for sure - just hope im not too late.

**Comment by AikonMishima** [comment]
This would indeed be a great option.

It'd definitely help with the NFT feature of [@LVLone](https://lvlone.softr.app/achievements), as they'll be 1/1 NFT's bound to an individual or a group's specific achievements.

My workaround was going to be to set the royalties to 100, so it wouldn't have made sense for people to sell their NFT.

**Comment by tijno** [comment]
Updated the original post with some more info after reviewing some of the pending NFT code.

@maebeam @lazynina @redpartyhat  Is this the right place to raise this? Or should I mention it in the relevant PRs for nifties?

**Reply by AikonMishima** [reply]
+1 for the updated CIP proposal.

Especially this:
“ Gift or transfer NFT to a user - but probably only allow this if this was not won in a Auction. Otherwise it would offer a way for users to sell NFTs without royalties.”

**Comment by jaydih** [comment]
I love this idea.

**Comment by comtechnet** [comment]
If indeed u are going to allow 2 "broad" categories of Bitclout NFTs 1) transferrable and 2) NON transferrable; ensure that these still comply with the erc721 (etc) standards) / it seems NON transferrable needs to "stay" within the Bitclout ecosystem until this feature is universal for all NFTs / thoughts?

**Reply by tijno** [reply]
I think BitClout NFTs currently are only within the BitClout ecosystem and dont follow ERC721 standard. ERC721 is a ETH based standard. It would be amazing if someone did make BitClout NFTs transferable between ERC721 and BitClout both ways!!

**Reply by comtechnet** [reply]
Looks like FLOW has "started" this movement / so other's will have to step up.  NFTs need to be able to be transferrable - just as "ART" is transferable (proven, reliable, verifiable, etc.) - I know the "physical item" behind some NFTs (where there is some kind of attachment/connection to a physical item - may be a little arduous to prove/verify). but the NFT itself should be completely transferrable and transportable.  Now that FLOW has done this for the first time, between FLOW's blockchain and Ethereum - other's will need to follow.  (IMHO)

**Comment by jsonpreet** [comment]
Great @tijno

**Comment by salilsethi** [comment]
Love the idea. I can think of several use cases related to ProsperClout here :)

**Comment by tijno** [comment]
Its a shame this requested change may not make it into the first release as per @diamondhands0 comment here:

https://github.com/bitclout/cips/pull/92#issuecomment-886298622

Its got huge support from the community and is easy to add.

> Not a hard thing to implement, but it's going to take a bit more thought from the UX side to make sure that this gets messaged to buyers without cluttering the look of an NFT in the feed.

The lock icon is not a good suggestion as the "has unlockable content" already uses it in the feed.

But could use the same mechanism, just use a different icon.

https://github.com/bitclout/frontend/blob/rph/nifties/src/app/feed/feed-post/feed-post.component.html#L442

```html
<span class="d-lg-inline-block d-none">|&nbsp;</span>
<span 
      class="cursor-pointer d-lg-inline-block d-block"
      (click)="tooltip.toggle();"
      matTooltipClass="global__mat-tooltip global__mat-tooltip-font-size"
      [matTooltip]="tradableTooltip"
      #tooltip="matTooltip"
>
  <i
    class="fas"
    [ngClass]="{
    'fa-store-slash': !IsTradable,
    'fa-store': IsTradable
    }"
  ></i>
  {{ IsTradable ? "Can be traded/sold" : "Can not be traded/sold"  }}
</span>
```

Using these icons

![20210726-PEs7bAgg](https://user-images.githubusercontent.com/69529928/126962769-aca53b91-6a68-4c40-9dc0-eb81934c00cb.png)
![20210726-QHjtC76f](https://user-images.githubusercontent.com/69529928/126962773-bd70b9c6-a5c6-4f67-b4ab-ed12c0b94aaa.png)

**Reply by tijno** [reply]
If NFT royalties are 100% the seller wont get any money from reselling.

So this kinda achieves the objective but doesnt completely stop it.

Could be an easy way to implement this - eg if Combined royalties are 100% then it can not be resold.

**Comment by andyboyd** [comment]
Really love the gifting side of this. The non resellable part is cool too, but the gifting would open up so many options. Very, very, super keen for that to be added.

Is there any chance this proposal could be split in two, since the two parts seem like they may be able to reach the benchmark for acceptance independently?

**Reply by tijno** [reply]
I guess the use cases I outlined requires both to be in place.

So splitting from my perspective doesnt make sense.

What uses do you see for gifting on its own ?

**Reply by andyboyd** [reply]
Well, one example could be if an app wanted to act like an exchange for fixed price NFTs, that could be achieved by acting as a middle man, NFT owners gifting their NFTs to an account owned by the app, and the app automating the bid/acceptance process to make it one click for the buyer with no further involvement needed from the seller. 

Without gifting, that would require the middle man to place bids and the seller to accept them, which could still be automated, but seems inefficient.

I fully understand that the non transferability requires gifting, but I don't think the same is true in reverse. I do think both features have a lot of merit though, and if you'd rather keep it all as one proposal, I totally support it in full.

**Comment by trevormil** [comment]
https://github.com/bitclout/cips/discussions/105

**Reply by tijno** [reply]
BidBadges is cool and for sure complimentary - and I think it would benefit for sure from these two proposed enhancements to BitClout NFTs

1 - marking NFTs on Bitclout as not for resale
2 - allowing users to transfer / move / gift NFTs to another account

**Comment by DarrylGold** [comment]
Here's my use case...

I would like to create an NFT. Say 100/100. And I would like to send them to people, just as a gift. They don't have to buy them. They can just have them. And they can even resell them. And any future sales on the NFTs still have a royalty which comes back to me.

I'm trying to understand what you are proposing and I am not clear if you mean that if you gift an NFT that it can't be resold or have a royalty. Are the features are mutually exclusive? Please clarify.

Thanks.

**Reply by tijno** [reply]
Proposing:
- gift nft that can not be resold
- sell nft that can not be resold
- gift nft that can be resold

so your usecase would ge supported

**Reply by DarrylGold** [reply]
Gift NFTs can be resold with royalties on resales?

**Reply by tijno** [reply]
They are 2 separate "features" proposed 

1 - the feature to set a NFT to not for relate
2 - the feature to be able to transfer (eg gift) NFT to another account

With both of these the above 3 options are possible including your request for Gifting an NFT that the reciever could then resell.

**Comment by F041** [comment]
That would help me with a project, even though i didn't think about the 100% royalties, that would do the trick as well
